### ISSUE TITLE
JavaScript files do not allow export default declarations

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#6794]
From http://stackoverflow.com/questions/35134618/typescript-1-8-es2015-imports-in-js-files

```js
export default "hello";
```

Gives the incorrect error:

```
hello.js(1,1): error TS8003: 'export=' can only be used in a .ts file.
```
